[docs] First draft for updated `README.md` (#1069)
single-cell-data/TileDB-SOMA: README.md

# TileDB-SOMA

[SOMA](https://github.com/single-cell-data/SOMA/tree/main) – for “Stack Of Matrices, Annotated” – is a flexible, extensible, and open-source API enabling access to data in a variety of formats. The driving use case of SOMA is for single-cell data in the form of annotated matrices where observations are frequently cells and features are genes, proteins, or genomic regions.

The TileDB-SOMA package is a C++ library with APIs in Python and R, using [TileDB
Embedded](https://github.com/TileDB-Inc/TileDB) to implement the
[SOMA specification](https://github.com/single-cell-data/SOMA/blob/main/abstract_specification.md).

Get started on using TileDB-SOMA:

* [Quick start](#quick-start).
* Python [documentation](#tiledb-soma) | [tutorials](#tiledb-soma). *Under development*.

> * R [documentation](#tiledb-soma) | [tutorials](#tiledb-soma). *Under development*.

Reviewer: I think they're supposed to link to the docpages that do not exist yet.

> * R [documentation](#tiledb-soma) | [tutorials](#tiledb-soma). *Under development*.

Author: Indeed will update once this is ready

## What can TileDB-SOMA do?

Intended to be used for single-cell data, TileDB-SOMA provides Python and R APIs to allow for storage and data access patterns at scale and for larger-than-memory operations:

* Create and write large volumes of data.
* Open and read data at low latency, locally and from the cloud.
* Query and access interconnected arrays efficiently and at low latency.

> TiledDB-SOMA provides interoperability with existing single-cell toolkits:

Author: Thanks!

* Load and create [anndata](https://anndata.readthedocs.io/en/latest/) objects.

> * Load and create [Seurat](https://anndata.readthedocs.io/en/latest/) objects. *Coming soon*.

Reviewer 2: ``
```suggestion
* Load and create [Seurat](https://satijalab.org/seurat/) objects. *Coming soon*.
```

> * Load and create [Seurat](https://anndata.readthedocs.io/en/latest/) objects. *Coming soon*.

Author: fixed

TiledDB-SOMA provides interoperability with existing Python or R data structures:

* From Python create PyArrow objects, SciPy sparse matrices, NumPy arrays, and pandas data frames.

> * From R create RArrow objects, Matrix sparse matrices, and R-base dense matrices.

Reviewer 2: ```suggestion
* From R create R Arrow objects, sparse matrices (via the [Matrix](https://cran.r-project.org/package=Matrix) package), and standard data frames and (dense) matrices.
```

## Community

* Please join the [TileDB Slack community](https://tiledb-community.slack.com/join/shared_invite/zt-ndq1ipwl-QcithaWG6j1BImtuQGSpag#/shared-invite/email) with dedicated channel `#genomics`
* Please join the [CZI Slack community](https://cziscience.slack.com/), with dedicated
channel `#cell-census-users`

## Quick start

### Documentation

The TileDB-SOMA doc-site (under development), contains the reference documentation and tutorials.

Reference documentation can also be accessed directly from Python `help(tiledsoma)` or R `help(package = "tiledbsoma")`.

### Main SOMA Objects

The capabilities of TileDB-SOMA lay on the different read, access, and query patterns that each of the main implementations of SOMA objects provide:

> * `DenseNDArray` a dense, N-dimensional array, with offset (zero-based) integer indexing on each dimension.

Reviewer: ```suggestion
* `DenseNDArray` is a dense, N-dimensional array, with offset (zero-based) integer indexing on each dimension.
```

> * `SparseNDArray` same as `DenseNDArray` but sparse, and it supports point indexing (disjoint index access)

Reviewer: ```suggestion
* `SparseNDArray` is the same as `DenseNDArray` but sparse, and supports point indexing (disjoint index access)
```

> * `DataFrame`  a multi-column table with a user-defined columns names and value types, with support for point indexing. 

Reviewer: ```suggestion
* `DataFrame` is a multi-column table with a user-defined columns names and value types, with support for point indexing. 
```

> * `Collection` a persistent container of named SOMA objects.

Reviewer: ```suggestion
* `Collection` is a persistent container of named SOMA objects.
```

* `Experiment` is a class that represents a single-cell experiment. It always contains two objects:
	* `obs`: a  `DataFrame` with primary annotations on the observation axis.

> 	* `ms`: a  `Collection` of measurements, each composed of `X` matrices and axis annotation matrices or data frames.

Reviewer 2: Let's add periods to all bullets or none

# TileDB-SOMA

[SOMA](https://github.com/single-cell-data/SOMA/tree/main) – for “Stack Of Matrices, Annotated” – is a flexible, extensible, and open-source API enabling access to data in a variety of formats. The driving use case of SOMA is for single-cell data in the form of annotated matrices where observations are frequently cells and features are genes, proteins, or genomic regions.

The TileDB-SOMA package is a C++ library with APIs in Python and R, using [TileDB
Embedded](https://github.com/TileDB-Inc/TileDB) to implement the
[SOMA specification](https://github.com/single-cell-data/SOMA/blob/main/abstract_specification.md).

Get started on using TileDB-SOMA:

* [Quick start](#quick-start).

> * Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html) | [tutorials](#tiledb-soma) (*Coming soon*). 

Reviewer 3: ```suggestion
* Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html) | [tutorials](#tiledb-soma)
```

this is no longer "coming soon". it is already live.

> * Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html) | [tutorials](#tiledb-soma). 
> * R [documentation](https://single-cell-data.github.io/TileDB-SOMA/) | [tutorials](#tiledb-soma).

Reviewer 3: ```suggestion
* Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html). 
* R [documentation](https://single-cell-data.github.io/TileDB-SOMA/).
```

Per our Slack sync :)

> * R [documentation](https://single-cell-data.github.io/TileDB-SOMA/) | [tutorials](#tiledb-soma) (*Coming soon*).

Reviewer 3: ```suggestion
* R [documentation](https://single-cell-data.github.io/TileDB-SOMA/) | [tutorials](#tiledb-soma).
```

this is no longer "coming soon". it is already live.

Get started on using TileDB-SOMA:

* [Quick start](#quick-start).

> * Python [documentation](https://tiledbsoma.readthedocs.io/en/latest/python-api.html) | [tutorials](#tiledb-soma) *Coming soon* . 

Reviewer 3: please rebase atop #1103
